src/main/java/org/apache/maven/plugins/jlink/JLinkMojo.java

```diff
@@ -510,6 +522,12 @@ private File createZipArchiveFromImage(File outputDirectory, File outputDirector
             throws MojoExecutionException {
         zipArchiver.addDirectory(outputDirectoryImage);
 
+        // configure for Reproducible Builds based on outputTimestamp value
+        Date lastModified = new MavenArchiver().parseOutputTimestamp(outputTimestamp);
+        if (lastModified != null) {
+            zipArchiver.configureReproducible(lastModified);
+        }
```

Comment:
Why is the null check needed here, but not in https://github.com/apache/maven-rar-plugin/commit/6ea9c738f089930b5f27b4e7796808412b1886b0/ ?

Reply:
because here it a method for Plexus Archiver, while in maven-rar-plugin it's a Maven Archiver (that only works on jars)